packages/ui/primitives/document-password-dialog.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogDescription,
+  DialogFooter,
+} from './dialog'; 
+
+import { Input } from './input'; 
+import { Button } from './button'; 
+
+type PasswordDialogProps = {
+  open: boolean;
+  onOpenChange: (_open: boolean) => void;
+  setPassword: (_password: string) => void;
+  handleSubmit: () => void;
+  isError?: boolean;
+}
+
+export const PasswordDialog = ({ open, onOpenChange, handleSubmit, isError, setPassword }: PasswordDialogProps) => {
+  return (
+    <Dialog open={open} onOpenChange={onOpenChange}>
+      <DialogContent className="max-w-md">
+        <DialogHeader>
+          <DialogTitle>Password Required</DialogTitle>
+          <DialogDescription>
```

Comment:
Can you maintain `This document is password protected. Please enter the password to view the document.` as the description and show the error below the input?

Much like what we do on the sign up page

Reply:
![image](https://github.com/documenso/documenso/assets/71957674/28affd70-8e22-4f2b-b1a0-482d0fab3d29)


Updated !